src/CMakeLists.txt

```diff
@@ -143,6 +143,17 @@ if(APPLE AND (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_
         list(REMOVE_DUPLICATES CMAKE_CXX_IMPLICIT_LINK_LIBRARIES)
 endif()
 
+# Check for base threading library
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package(Threads REQUIRED)
+if(${CMAKE_THREAD_LIBS_INIT} MATCHES "-pthread")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_THREAD_LIBS_INIT}")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_THREAD_LIBS_INIT}")
+elseif(CMAKE_THREAD_LIBS_INIT)
+  list(APPEND LIBS ${CMAKE_THREAD_LIBS_INIT})
+endif()
+
```

Comment:
Later on, we look for pthreads using our own findpthreads.cmake module.
https://github.com/mazhe/darktable/blob/8787f60969496381032b5bb1a1e0d8bbdf28e910/src/CMakeLists.txt#L240
https://github.com/mazhe/darktable/blob/8787f60969496381032b5bb1a1e0d8bbdf28e910/cmake/modules/FindPThread.cmake
I guess it should be dropped.